iocore/net/quic/QUICConnection.h

```diff
@@ -65,4 +65,15 @@ class QUICConnection : public QUICPacketTransmitter, public QUICFrameTransmitter
   virtual QUICPacketNumber largest_acked_packet_number()    = 0;
   virtual void handle_received_packet(UDPPacket *packeet)   = 0;
   virtual bool is_closed()                                  = 0;
+
+  bool
+  shouldDestroy()
+  {
+    return this->refcount() == 0;
+  }
+
+  int in_closed_queue = 0;
```

Comment:
`shouldDestroy()` and these close queue stuff are for management of NetVConnection, right? If so, they should be in `QUICNetVConnection`.

(`NetVConnectionContext_t` is the last piece that still rely on NetVC stuff. I'm going to generalize it later.)

Reply:
Sorry, I want to split the qvc into 2 part. 1: qvc 2: qc.

we are going to free qvc into ET_NET and free qc into ET_UDP. 
QUICClosedConCollector are used for collect the qc which closed , but unfortunately, it is not clearly enough to split it out.

Reply:
there are some bad code in QUICClosedConCollector like `qvc = static_cast<QUICNetVConnection>(qc)`

Reply:
Well, if I understand correctly, a conclusion at the QUIC hakcathon was that we keep a pure interface for testing and QUIC applications, and introduce another class for connection management like this.

Let's say you are creating a network application. You don't need to know how kernel manages TCP connection state, right? Usually, you need only `connect()`, `read()`, `write()`, `shutdown()` and `close()`. Even if kernel changed how it manages connection state, applications wouldn't be affected.

So the concept, splitting QNetVC into multiple parts,  is fine with me. If that is how ATS manages connections, maybe we can introduce an interface like this:

```
class ATSManagedConnection
{
  virtual bool shouldDestroy() = 0;
  virtual void destroy() = 0;
}
```
and QUICNetVConnection can implement the both this and QUICConnection.
```
class QUICNetVConnection extends UnixNetVConnection, ATSManagedConnection, QUICConnection
{
  ...
}
```


I'm not sure these two functions satisfy all situation, but I think it can be used not only QUICNetVC but also UnixNetVC if they both are managed with the same concept (so probably UnixNetVConnection should extends the interface class).